libnmstate/nm/sriov.py

```diff
@@ -0,0 +1,73 @@
+#
+# Copyright (c) 2019 Red Hat, Inc.
+#
+# This file is part of nmstate
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Lesser General Public License as published by
+# the Free Software Foundation, either version 2.1 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public License
+# along with this program. If not, see <https://www.gnu.org/licenses/>.
+#
+
+from libnmstate.nm import connection as nm_connection
+from libnmstate.nm import nmclient
+from libnmstate.schema import Ethernet
+
+
+def create_setting(iface_state, base_con_profile):
+    sriov_config = iface_state.get(Ethernet.CONFIG_SUBTREE, {}).get(
+        Ethernet.SRIOV_SUBTREE
+    )
+    if not sriov_config:
+        return sriov_config
+
+    sriov_setting = _get_current_sriov_setting(base_con_profile)
+    if not sriov_setting:
+        sriov_setting = nmclient.NM.SettingSriov.new()
+
+    if sriov_config.get(Ethernet.SRIOV.TOTAL_VFS):
+        sriov_setting.props.total_vfs = sriov_config[Ethernet.SRIOV.TOTAL_VFS]
+
+    return sriov_setting
+
+
+def _get_current_sriov_setting(base_con_profile):
```

Comment:
This could probably be a generic function that takes the `base_con_profile` parameter and the constant for the setting. Other code in the nm directory could then re-use this.

Reply:
This should be another PR right?

Reply:
yes, at least making the other code use the same function. You can already introduce it here and use it for this code IMHO. It might also be useful for the team implementation.

Reply:
Done thanks!